ingester/plugins/weather_prediction_plugin.go

```diff
@@ -0,0 +1,325 @@
+package main
+import (
+	"fmt"
+	"github.com/gtfierro/xboswave/ingester/types"
+	xbospb "github.com/gtfierro/xboswave/proto"
+)
+
+func has_device(msg xbospb.XBOS) bool {
+	return msg.XBOSIoTDeviceState.WeatherPrediction != nil
+}
+
+// This contains the mapping of each field's value to the unit
+var device_units = map[string]string{
+	"time":	"seconds",
+	"icon":	"",
+	"precipIntensity":	"inches per hour",
+	"precipIntensityError":	"",
+	"precipProbability":	"",
+	"temperature":	"F",
+	"apparentTemperature":	"F",
+	"dewPoint":	"F",
+	"humidity":	"",
+	"pressure":	"millibars",
+	"windSpeed":	"miles per hour",
+	"windGust":	"miles per hour",
+	"windBearing":	"degree",
+	"cloudCover":	"",
+	"uvIndex":	"",
+	"visibility":	"miles",
+	"ozone":	"Dobson",
+}
+
+
+func Extract(uri types.SubscriptionURI, msg xbospb.XBOS, add func(types.ExtractedTimeseries) error) error {
+	if msg.XBOSIoTDeviceState != nil {
+
+		if has_device(msg) {
+            var step int
+            step = 0
+
+            //Iterate through each hour of prediction from current to 48 hours from current
+			for _, _prediction := range msg.XBOSIoTDeviceState.WeatherPrediction.Predictions {
+                //This prediction contains all of the fields that were present in WeatherCurrent message
+                //There is one for each hour that is retrieved from the DarkSky API
+				prediction := _prediction.Prediction
+
+                //This will contain all the information necessary to send one prediction for one hour out of 0-48
+				var extracted types.ExtractedTimeseries
+				var name string
+
+                //This is the xbos time
+				time := int64(msg.XBOSIoTDeviceState.Time)
+
+
+                //This is the time that is being put into influx as the timestamp
+				extracted.Times = append(extracted.Times, time)
+
+            	if prediction.PrecipIntensity != nil {
+                    //This will be the value that is put into a field in Influx
+            		extracted.Values = append(extracted.Values, float64(prediction.PrecipIntensity.Value))
```

Comment:
Have you thought about using more generic functions like we have above for this? For example [the existing weather plugin](https://github.com/gtfierro/xboswave/blob/master/ingester/plugins/weather_station_plugin.go#L82-L88). That should reduce duplicated code